src/panels/config/automation/action/types/ha-automation-action-device_id.ts

```diff
@@ -135,8 +135,11 @@ export class HaDeviceAction extends LitElement {
     }
   }
 
-  protected updated(changedPros) {
-    const prevAction = changedPros.get("action");
+  protected updated(changedProps) {
+    if (this.hass && changedProps.has("hass") && !changedProps.get("hass")) {
+      this.hass.loadBackendTranslation("device_automation");
```

Comment:
Cant we do this in `firstUpdated`?

Reply:
`_getCapabilities` needs `hass` too, and that is done in `firstUpdated`

Reply:
I am never confident I can understand when hass can be assumed to be available and when it isn't. Is there any kind of general rule I can follow?

In the past when I assume hass will be available it always turns out to be some kind of race condition and it sometimes isn't there when expected.

Reply:
normally for `hass` it will be available as `hass` is setup before any of the UI is shown. But in some cases it can be problematic because the parent will not pass it or something, but normally not be an issue...